sklearn/model_selection/_classification_threshold.py

```diff
@@ -0,0 +1,895 @@
+from collections.abc import MutableMapping
+from inspect import signature
+from numbers import Integral, Real
+
+import numpy as np
+
+from ..base import (
+    BaseEstimator,
+    ClassifierMixin,
+    MetaEstimatorMixin,
+    _fit_context,
+    clone,
+)
+from ..exceptions import NotFittedError
+from ..metrics import (
+    check_scoring,
+    get_scorer_names,
+    make_scorer,
+    precision_recall_curve,
+    roc_curve,
+)
+from ..metrics._scorer import _BaseScorer
+from ..utils import _safe_indexing
+from ..utils._param_validation import HasMethods, Interval, RealNotInt, StrOptions
+from ..utils._response import _get_response_values_binary
+from ..utils.metadata_routing import (
+    MetadataRouter,
+    MethodMapping,
+    _raise_for_params,
+    process_routing,
+)
+from ..utils.metaestimators import available_if
+from ..utils.multiclass import type_of_target
+from ..utils.parallel import Parallel, delayed
+from ..utils.validation import (
+    _check_method_params,
+    _num_samples,
+    check_consistent_length,
+    check_is_fitted,
+    indexable,
+)
+from ._split import StratifiedShuffleSplit, check_cv
+
+
+def _threshold_scores_to_class_labels(y_score, threshold, classes, pos_label):
+    """Threshold `y_score` and return the associated class labels."""
+    if pos_label is None:
+        map_thresholded_score_to_label = np.array([0, 1])
+    else:
+        pos_label_idx = np.flatnonzero(classes == pos_label)[0]
+        neg_label_idx = np.flatnonzero(classes != pos_label)[0]
+        map_thresholded_score_to_label = np.array([neg_label_idx, pos_label_idx])
+
+    return classes[map_thresholded_score_to_label[(y_score >= threshold).astype(int)]]
+
+
+class _ContinuousScorer(_BaseScorer):
+    """Scorer taking a continuous response and output a score for each threshold."""
+
+    def __init__(self, score_func, sign, kwargs, n_thresholds, response_method):
+        super().__init__(
+            score_func=score_func,
+            sign=sign,
+            kwargs=kwargs,
+            response_method=response_method,
+        )
+        self._n_thresholds = n_thresholds
+
+    @classmethod
+    def from_scorer(cls, scorer, response_method, n_thresholds, pos_label):
+        """Create a continuous scorer from a normal scorer."""
+        # add `pos_label` if requested by the scorer function
+        scorer_kwargs = {**scorer._kwargs}
+        signature_scoring_func = signature(scorer._score_func)
+        if (
+            "pos_label" in signature_scoring_func.parameters
+            and "pos_label" not in scorer_kwargs
+        ):
+            if pos_label is None:
+                # Since the provided `pos_label` is the default, we need to
```

Comment:
Is there a way to make this more consistent in a backward compatible way?

Reply:
I don't think that we can do much more in this PR. However, we should address the long term issue related to the meaning of `pos_label`: https://github.com/scikit-learn/scikit-learn/issues/10010